src/appindexing.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+goog.provide('appindexing');
+
+goog.require('goog.json');// jshint unused:false
+goog.require('utils');
+
+
+appindexing.state = {};
+appindexing.state['androidAppIndexingTagsPresent'] = false;
+appindexing.state['iOSAppIndexingTagsPresent'] = false;
+appindexing.state['androidDetailsComplete'] = false;
+appindexing.state['iOSDetailsComplete'] = false;
+appindexing.state['addediOSTagDom'] = false;
+appindexing.state['addedAndroidTagDom'] = false;
+
+appindexing.checkForAppIndexingTags = function() {
```

Comment:
Can you add some quick high level comments on what these functions do? For example, this one:
```
// Will scan through all elements the tag name link and update the href value for any which contain
// "ios-app" or "android-app" to include Branch tracking parameters.